### ISSUE TITLE
snom phone does not show caller name/number in call

### ISSUE THREAD

**Issue by hehol** [amooma/GS5#68]
When user A calls user B, the name and number of user A are being shown in the ringer screen of user B's phone. However, as soon as the call is being answered, the name and number are being replaced with the string "Call" plus the account id of the calling phone.

Example:

![screen-ring](https://f.cloud.github.com/assets/239507/46137/ab808d08-5825-11e2-9f6d-75c006316333.png)
while ringing

![screen-call](https://f.cloud.github.com/assets/239507/46139/ed5b056e-5825-11e2-833f-4038a855ed13.png)
in call

**Comment by spag** [comment]
I'm afraid it's some strange Snom/FreeSWITCH thing we can't do much about.

**Comment by wintermeyer** [comment]
@spag Could this be the solution? http://wiki.freeswitch.org/wiki/Variable_effective_caller_id_name

**Comment by hehol** [comment]
@wintermeyer No, effective_caller_id_name only affects the caller-id during the origination phase of the channel. It will be overwritten once the call has been fully bridged. effective_caller_id_name/number are already being used to override the caller-id name and number in SipCall.fork(). This is a bit tricky in FS because the name and number will fall back to the data of the SIP peer in subsequent messages of the transaction.

Using http://wiki.freeswitch.org/wiki/Variable_ignore_display_updates will prevent that display update (COLP) from being sent.

I just tried this quick hack:

``` diff
diff --git a/misc/freeswitch/scripts/dialplan/sip_call.lua b/misc/freeswitch/scripts/dialplan/sip_call.lua
index 57f92c6..0db4125 100644
--- a/misc/freeswitch/scripts/dialplan/sip_call.lua
+++ b/misc/freeswitch/scripts/dialplan/sip_call.lua
@@ -107,6 +107,7 @@ function SipCall.fork(self, destinations, arg )
       if not call_waiting then
         destinations[index].numbers = sip_account:phone_numbers();

+       table.insert(origination_variables, "ignore_display_updates=true");
         if not arg.callee_id_name then
           table.insert(origination_variables, "effective_callee_id_name='" .. sip_account.record.caller_name .. "'");
         end
```

and it fixes the problem, so ignore_display_updates seems to be the way to go indeed.

**Comment by wintermeyer** [comment]
Please add a note that you assign the rights for this patch to AMOOMA GmbH. After that we can use it. Thanks!

**Comment by hehol** [comment]
Feel free to use the patch. Just keep in mind that it's a quick hack and probably not the right place in the code to set the variable.

**Comment by spag** [comment]
I we really need that option it should be made available in "dialplan.ini". 
If we disable it in the code for good, some people would complain to have lost sensible functionality on phones not affected by this issue (e.g. Polycom)

**Comment by spag** [comment]
@hehol Which firmware Version do you use?

**Comment by hehol** [comment]
@spag I tried 8.4.35 and 8.7.3.15 with the same result.

**Comment by spag** [comment]
One disturbing fact about Snom phones seems to be that you cannot reliably set them back to factory defaults.
e.g. different Snom 821 phones  (both fw. 8.4.35) show different settings (end behave differently) right after performing a factory reset.

**Comment by spag** [comment]
On a Snom 821 with FW 8.4.35:
![snom821_1](https://f.cloud.github.com/assets/230554/49555/76e01978-596b-11e2-8306-d67604631592.png)
![snom821_2](https://f.cloud.github.com/assets/230554/49556/7d57bfa4-596b-11e2-89f1-4768f05d8aee.png)

The labels are not correct (both lines show callee data) but the information displayed seems to be legit. 
One could probably tinker with the phone firmware to change the labels :)

**Comment by hehol** [comment]
@spag your screenshots have been taken from display of the caller, right? My screenshots show the display of the callee.

I don't believe this is an issue with the factory reset of the snom phones. To be absolutely sure, you might want to erase the settings partition from the bootloader if you are using a 7xx or 8xx phone.

**Comment by spag** [comment]
fixed in commit 70ef0ee8e668b0f620ec74afc8347e43bcfebb87
